File: frontend/src/types/task.ts
/**
 * タスク関連の TypeScript 型定義。
 *
 * Task、TaskSummary インターフェースを定義し、
 * Claude Code Agent Teams のタスク管理を型安全に行います。
 *
*/

/**
 * タスク詳細情報を表すインターフェース。
 *
 * タスクID、件名、ステータス、担当者、依存関係、メタデータを持ちます。
 * pending/in_progress/completed/deleted のステータスを取ります。
 */
export interface Task {
  id: string;
  subject: string;
  description?: string;
  activeForm: string;
  status: 'pending' | 'in_progress' | 'completed' | 'deleted';
  owner?: string;
  blocks: string[];
  blockedBy: string[];
  metadata?: Record<string, unknown>;
  teamName?: string;
}

/**
 * タスク一覧表示用のサマリーインターフェース。
 *
 * タスクID、件名、ステータス、担当者、依存タスク数を持ちます。
 * 一覧画面での高速表示用に最適化されています。
 */
export interface TaskSummary {
  id: string;
  subject: string;
  status: 'pending' | 'in_progress' | 'completed' | 'deleted';
  owner?: string;
  blockedCount: number;
  teamName?: string;
}

File: backend/app/models/task.py
"""タスク関連の Pydantic データモデル。

Task、TaskSummary、TaskUpdate の3つのモデルを定義し、
チーム内のタスク管理と状態追跡をサポートします。

"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class Task(BaseModel):
    """タスクの完全なデータモデル。

    タスクのID、件名、説明、ステータス、担当者、依存関係（blocks/blockedBy）、
    メタデータを含みます。ステータスは pending/in_progress/completed/deleted です。


    """

    id: str
    subject: str
    description: Optional[str] = ""
    activeForm: str = ""
    status: str  # pending, in_progress, completed, deleted, stopped
    owner: Optional[str] = None
    blocks: list[str] = []
    blockedBy: list[str] = []
    metadata: dict = {}


class TaskSummary(BaseModel):
    """タスク一覧表示用のサマリーモデル。

    タスクID、件名、ステータス、担当者、依存タスク数、チーム名を含みます。
    一覧画面での高速表示に最適化された軽量モデルです。


    """

    id: str
    subject: str
    status: str
    owner: Optional[str] = None
    blockedCount: int
    teamName: Optional[str] = None


class TaskUpdate(BaseModel):
    """タスク更新イベントのデータモデル。

    タスクID、チーム名、新しいステータス、変更前ステータス、タイムスタンプを含みます。
    ポーリングによるデータ取得時に変更検知に使用します。


    """

    taskId: str
    teamName: str
    status: str
    previousStatus: Optional[str] = None
    timestamp: datetime
